### ISSUE TITLE
Is the loss function really correct ?

### ISSUE THREAD

**Issue by rotalex** [marvis/pytorch-yolo3#26]
I cannot understand the following two lines of code in region_loss.py, line 94
`iou = bbox_iou(gt_box, pred_box, x1y1x2y2=False) # best_iou`
`tconf[b][best_n][gj][gi] = iou`

Why the confidence is set to the iou between predicted and ground truth and not the iou between ground truth and anchor boxes ?
Shouldn't it be something like:
`iou = bbox_iou(gt_box, anchor_box, x1y1x2y2=False) # best_iou`
Am I missing something ?

**Comment by alexandru-dinu** [comment]
I am also having this concern.

**Comment by KrnTneja** [comment]
As far as I know, this is correct as the loss is calculated between the ground truth and the predicted box. If you think about it IoU between the ground truth box and a specific anchor box for a image will always be fixed. This IoU is only helpful in deciding the grid cell to which ground truth box will be matched.
